app/views/user/products/_product_form.html.haml

```diff
@@ -0,0 +1,190 @@
+            = "Trans fat - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "trans_fat", name: "product[trans_fat]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Fiber - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "fiber", name: "product[fiber]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Sugar - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "sugar", name: "product[sugar]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Sodium - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "sodium", name: "product[sodium]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Vitamin A - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "vitamin_A", name: "product[vitamin_A]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Vitamin C - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "vitamin_C", name: "product[vitamin_C]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Calcium - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "calcium", name: "product[calcium]", min: "0", value: nil }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Iron - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ type: "number", id: "iron", name: "product[iron]", min: "0", value: nil }
+
+    .flex.flex-col.content-center.w-full{ class: "tab" }
+        %label.text-xl.flex.flex-col.mb-4
+            = "Allergens - Optional"
+            %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ id: "allergens", class: 'allergen-autocomplete', value: nil }
+            #allergens-anchor.w-full.flex.flex-wrap.gap-2
+        /
+            %label.text-xl.flex.flex-col.mb-4
+                = "Weight Units - Optional"
+                %input.text-l.border.border-slate-700.rounded-md.p-1.bg-slate-50.mt-2{ id: "weight_units", name: "product[weight_units[]]", value: nil }
+
+    .flex.flex-row.justify-end.items-center.w-full.mb-2
+        %button.bg-zinc-600.rounded-md.text-white.px-4.p-2{ id: "previousButton", onClick: "tabShift(-1)", type: "button" }
+            = "Previous"
+        %button.bg-lime-600.rounded-md.text-white.px-4.p-2.ml-2{ id: "nextButton", onClick: "tabShift(1)", type: "button" }
+            = "Next"
+    
+    .flex.flex-row.justify-center.w-full 
+        - (1..5).each do |step|
+            %span{ class: "step" }
+
+:javascript
```

Comment:
Enclose all javascript in a local scope to prevent issues from rewriting the document by the FormController